pkg/cmd/experimental/router/router.go

```diff
@@ -191,23 +227,9 @@ func NewCmdRouter(f *clientcmd.Factory, parentName, name string, out io.Writer)
 								Template: &kapi.PodTemplateSpec{
 									ObjectMeta: kapi.ObjectMeta{Labels: label},
 									Spec: kapi.PodSpec{
-										Containers: []kapi.Container{
-											{
-												Name:  "router",
-												Image: image,
-												Ports: ports,
-												Env:   env.List(),
-												LivenessProbe: &kapi.Probe{
-													Handler: kapi.Handler{
-														TCPSocket: &kapi.TCPSocketAction{
-															Port: kutil.IntOrString{
-																IntVal: ports[0].ContainerPort,
-															},
-														},
-													},
-												},
-											},
-										},
+										Containers:  getRouterPodContainers(cfg, env),
+										Volumes:     getFailoverContainerVolumes(),
+										HostNetwork: true,
```

Comment:
Does the kubelet see the host network setting?  You can query the kubelet directly to see if it's getting the right response.